substrate/frame/session/src/lib.rs

```diff
@@ -400,6 +401,10 @@ pub mod pallet {
 	#[pallet::without_storage_info]
 	pub struct Pallet<T>(_);
 
+	/// A simple identifier for session keys hold reason.
+	#[derive(codec::Encode, codec::Decode, codec::MaxEncodedLen, Debug, PartialEq, Eq, Clone)]
+	pub struct SessionKeysHoldReason;
```

Comment:
```suggestion
	pub enum SessionHoldReason {
	    #[codec(index = 0)]
	    Keys
};
```

This will be forward compatible in case we add more "reasons".